network/p2p/p2p_decided.go

```diff
@@ -26,6 +26,15 @@ func (n *p2pNetwork) BroadcastDecided(topicName []byte, msg *proto.SignedMessage
 
 	n.logger.Debug("Broadcasting decided message", zap.String("lambda", string(msg.Message.Lambda)), zap.Any("topic", topic), zap.Any("peers", topic.ListPeers()))
 
+	// publishing on main topic as well
```

Comment:
@amir-blox should this be in network? why not in the node logic?

Reply:
After `ExecuteDuty`?  it will happen after signature collection.. 
Other places could be duty_execution or IBFT but then I need to restore the message to send